components/usage-api/usage/v1/usage.proto

```diff
@@ -15,6 +15,66 @@ service UsageService {
 
     // GetCostCenter retrieves the spending limit with its associated attributionID
     rpc GetCostCenter(GetCostCenterRequest) returns (GetCostCenterResponse) {}
+
+    // Triggers reconciliation of usage with ledger implementation.
+    rpc ReconcileUsageWithLedger(ReconcileUsageWithLedgerRequest) returns (ReconcileUsageWithLedgerResponse) {}
+
+    // Creates, or updates, a Usage record
+    rpc UpsertUsage(UpsertUsageRequest) returns (UpsertUsageResponse) {}
+}
+
+enum UsageKind {
+    USAGE_KIND_UNKNOWN = 0;
+    USAGE_KIND_WORKSPACE_INSTANCE = 1;
+    USAGE_KIND_INVOICE = 2;
+}
+
+message Usage {
+    // UUID identifier of the Usage event. Required.
+    string id = 1;
+
+    // Attribution ID for the usage record. Required.
+    string attribution_id = 2;
+
+    // Description of the Usage event for humans.
+    string description = 3;
+
+    // Positive, or negative amount of credits for the Usage event.
+    // Credits are accepted with full precision, but returned with two decimal point precision.
+    double credits = 4;
+
+    // effective_time is the time when this record has taken effect.
+    google.protobuf.Timestamp effective_time = 5;
```

Comment:
As you know we have the (somewhat bad) habit of storing time as an iso string on application level fields. Which is why I did the same for `effectiveTime` on the DB level.

Reply:
Yep, we'd of course continue to store it as a string on the DB level but on the API level, we get a number of benefits from using the `google.protobuf.Timestamp`. It has good mappings to standard time objects for most languages and provides a fairly standard way to specify the timestamp on the API level.

But can change this into a string for the API as well, it just means an extra layer of parsing on the API (into time, and then again into a VarcharTime)